lighthouse-core/gather/fetcher.js

```diff
@@ -180,21 +180,29 @@ class Fetcher {
       }, options.timeout);
     });
 
-    const fetchStreamPromise = this._loadNetworkResource(url);
-    const stream = await Promise.race([fetchStreamPromise, timeoutPromise])
+    const responsePromise = this._loadNetworkResource(url);
+
+    /** @type {{stream: LH.Crdp.IO.StreamHandle|null, status: number|null}} */
+    const response = await Promise.race([responsePromise, timeoutPromise])
       .finally(() => clearTimeout(timeoutHandle));
 
-    return await this._readIOStream(stream, {timeout: options.timeout - (Date.now() - startTime)});
+    if (response.stream && response.status && response.status >= 200 && response.status <= 299) {
+      const content = await this._readIOStream(response.stream, {
+        timeout: options.timeout - (Date.now() - startTime),
+      });
+      return {content, status: response.status};
+    }
+    return {content: null, status: response.status};
```

Comment:
more for fun than suggestion, but I'm gathering data on personal approaches. do you find the following easier or harder to read? :)

```suggestion
    const isOK = response.status && response.status >= 200 && response.status <= 299
    if (!response.stream || !isOK) return {content: null, status: response.status};
    
    const timeout = options.timeout - (Date.now() - startTime);
    const content = await this._readIOStream(response.stream, {timeout});
    return {content, status: response.status};
```

Reply:
I think your suggestion is easier to read.